docs/Custom_Device_Tips_and_Tricks.md

```diff
@@ -0,0 +1,271 @@
+
+**ActionVIOnPaste**
+Executes when a user pastes a custom device item. This template helps create action VIs that check channel properties. For example, if the user pastes a page that configures a target, you can create an action VI to ensure that the new page does not attempt to reconfigure the target.
+
+You can also customize this template to prompt a user to enter new values for the pasted item. For example, if a user pastes a page that will conflict with existing pages, you can prompt the user to enter new values for the page.
+
+The template has the following unique parameters.<br />
+
+•	Ptr in—Reference to the custom device item whose XML declaration calls this action VI.<br />
+•	Parent—Reference to the parent of the custom device item whose XML declaration calls this action VI.<br />
+•	All Ptrs—Array of references to the items the user pasted. You can only select one item to copy. This array only contains one reference that matches the Ptr in reference.<br />
+
+**ActionVIOnCompile**
+Executes when VeriStand compiles the system definition file.
+If you deploy the system definition, then undeploy it, and then redeploy it without making changes, this template does not execute because the system definition does not recompile.
+
+You can customize this template to finish configuring your hardware. The system definition file compiles when a user deploys the system definition, so you can configure your hardware based on the final settings from the system definition.
+
+You can also customize the template to quickly gather host-side settings. For example, often the custom device RT Engine VI uses properties set in the system definition. You can customize this template to read the values on the host side, which is much faster than reading them from the real-time target. You can then gather the properties into a single cluster, convert that cluster to a data variant, and write the variant as a single item property.
+
+### Adding Toolbar buttons
+
+A *Toolbar button* appears in the toolbar of **System Explorer**. These buttons only appear when displaying the configuration page associated with the button.
+
+Within the **&lt;Page&gt;** tags for an item, you can use the **&lt;ButtonList&gt;** tag to configure the toolbar buttons that appear with the item's configuration page. Each **&lt;Button&gt;** must include a unique **&lt;ID&gt;** string that identifies the button. The toolbar button displays by default.
+
+However, in each page VI, you can use the Disable Dynamic Button VI and the Enable Dynamic Button VI to dynamically disable and enable a button for that page based on its unique ID. These VIs are useful when you want the toolbar button to appear only when certain conditions are true.
+
+These VIs are located in the labview\vi.lib\NI VeriStand\Custom Device API directory.
+
+The following is an example framework you can use to implement a toolbar button.
+
+![](images/Custom_Device_Dynamic_Button_Framework.JPG)<br />
```

Comment:
For whatever reason the picture has some red underline. And it looks a bit confusing. In the original pdf it looks ok.

How it renders now:
![image](https://user-images.githubusercontent.com/62251693/135117068-bed7c185-717e-4dae-97ac-7be147d93351.png)

How it should look like: 
![image](https://user-images.githubusercontent.com/62251693/135117837-32e89216-e2b1-4c50-8d28-7dad58672990.png)